library/src/webapp/js/lang-datepicker/lang-datepicker.js

```diff
@@ -1,2004 +1,377 @@
-					localDTPicker.datetimepicker("setDate", localDate);
-					// At this point we can go and ask the picker what it has rounded the time to.
-					localDate = localDTPicker.datetimepicker("getDate");
-					// Handle duration, which relies on the datepicker being set
-					handleDuration();
-				}
-
-				// if the picker is reliant in hidden fields, set them
-				setHiddenFields(localDate);
-			}
-		};
-
-		/**
-		 * Init the app by making a request for a language file.
-		 * Failing the locale, we look for the base language
-		 */
-         var userLocale = 'en-US';
-         var userLang = 'en';
-         if(window.top.sakai){
-             if(window.top.sakai.locale){
-                 if(window.top.sakai.locale.userLocale){
-                     userLocale = window.top.sakai.locale.userLocale.replace("_", "-");
-                 }
-                 if(window.top.sakai.locale.userLanguage){
-                     userLang = window.top.sakai.locale.userLanguage;
-                 }
-             }
-         }
-
-         // First try the full locale (fr-CA) then try just the language (fr)
-         if (typeof $.datepicker.regional[userLocale] !== "undefined") {
-           $.datepicker.setDefaults($.datepicker.regional[userLocale]);
-         } else if (typeof $.datepicker.regional[userLang] !== "undefined") {
-           $.datepicker.setDefaults($.datepicker.regional[userLang]);
-         } else {
-           $.datepicker.setDefaults($.datepicker.regional[""]);
-         }
-
-         // Load the localization for the timepicker if the timepicker is being used
-         if (typeof $.timepicker !== "undefined") {
-           if (userLocale == "en-US") {
-             // Special case for the USA and AM/PM
-             $.timepicker.setDefaults({timeFormat: "hh:mm tt"});
-           } else if (typeof $.timepicker.regional[userLocale] !== "undefined") {
-             $.timepicker.setDefaults($.timepicker.regional[userLocale]); 
-           } else if (typeof $.timepicker.regional[userLang] !== "undefined") {
-             $.timepicker.setDefaults($.timepicker.regional[userLang]); 
-           } 
-           else {
-             $.timepicker.setDefaults($.timepicker.regional[""]); 
-           }
-         }
-
-         init();
-         // Save the time in ms this was created
-         $.datepicker._initTime = Date.now();
-	};
-
-	// A really lightweight plugin wrapper around the constructor, 
-	// preventing against multiple instantiations
-	$.fn[pluginName] = function(options) {
-		return this.each(function() {
-			if(!$.data(this, 'plugin_' + pluginName)) {
-				$.data(this, 'plugin_' + pluginName, new Plugin(this, options));
-			}
-		});
+      
+      // Invalid date
+      return null;
+    }
+    
+    // Format a date for the input field
+    static formatForInput(date, useTime = true) {
+      if (!date) return "";
+      
+      date = DateHelper.normalize(date);
```

Comment:
You could use toTemporalInstant(date, { smallestUnit: "minute" }).toString() here.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal/Instant/toString

Reply:
Again, from https://www.npmjs.com/package/temporal-polyfill